File: src/test/java/org/cloudfoundry/credhub/config/VersionProviderTest.java
package org.cloudfoundry.credhub.config;

import org.cloudfoundry.credhub.util.ResourceReader;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(JUnit4.class)
public class VersionProviderTest {
  @Test
  public void currentVersion_returnsTheCurrentVersion() throws Exception {
    final ResourceReader resourceReader = mock(ResourceReader.class);
    when(resourceReader.readFileToString("version")).thenReturn("test version");

    final VersionProvider subject = new VersionProvider(resourceReader);

    assertThat(subject.currentVersion(), equalTo("test version"));
  }

  @Test
  public void currentVersion_whenTheVersionHasExtraneousWhitespace_trimsTheWhitespace() throws Exception {
    final ResourceReader resourceReader = mock(ResourceReader.class);
    when(resourceReader.readFileToString("version")).thenReturn("   test version   ");

    final VersionProvider subject = new VersionProvider(resourceReader);

    assertThat(subject.currentVersion(), equalTo("test version"));
  }
}

File: components/string-utilities/src/test/kotlin/org/cloudfoundry/credhub/config/VersionProviderTest.kt
package org.cloudfoundry.credhub.config

import org.cloudfoundry.credhub.utils.ResourceReader
import org.cloudfoundry.credhub.utils.VersionProvider
import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`

class VersionProviderTest {
    @Test
    @Throws(Exception::class)
    fun currentVersion_returnsTheCurrentVersion() {
        val resourceReader = mock(ResourceReader::class.java)
        `when`(resourceReader.readFileToString("version")).thenReturn("test version")

        val subject = VersionProvider(resourceReader)

        assertThat(subject.currentVersion(), equalTo("test version"))
    }

    @Test
    @Throws(Exception::class)
    fun currentVersion_whenTheVersionHasExtraneousWhitespace_trimsTheWhitespace() {
        val resourceReader = mock(ResourceReader::class.java)
        `when`(resourceReader.readFileToString("version")).thenReturn("   test version   ")

        val subject = VersionProvider(resourceReader)

        assertThat(subject.currentVersion(), equalTo("test version"))
    }
}
